### PR TITLE
Fix average consumption calculation

### PR DESCRIPTION
**Pull request by lomamech**
- Improvement of the way to obtain the number of months necessary to
  calculate the average monthly consumption
- Implement integration test for Automaticallu calculated key values

### CODE REVIEW

**Comment by lomamech** [comment]
@mbayopanda  I need a review

**Comment by lomamech** [comment]
@mbayopanda  all checks have passed,

File: server/controllers/stock/index.js [hunk 1]
```diff
@@ -382,9 +382,9 @@ function listLotsDepot(req, res, next) {
  */
 function listInventoryDepot(req, res, next) {
   const params = req.query;
-  params.monthAverageConsumption = req.session.enterprise.settings.month_average_consumption;
+  const monthAverageConsumption = req.session.enterprise.settings.month_average_consumption;
```
Comment (mbayopanda) [inline server/controllers/stock/index.js:386]: 👍

**Review by mbayopanda — COMMENTED** [review]
@lomamech la fonction `processStockConsumptionAverage` est aussi utilisée dans la fonction `getLotsDepot`

File: test/integration/stock/stock.js [hunk 2]
```diff
@@ -183,4 +183,31 @@ describe('(/stock/) The Stock HTTP API', () => {
       })
       .catch(helpers.handler),
   );
+
+  it(
+    `/stock/inventories/depots Get Inventories in Stock By Depot`,
```
Comment (jniles) [inline test/integration/stock/stock.js:188]: 👍 🎆

**Comment by lomamech** [comment]
@mbayopanda  All Checks have passed

**Comment by mbayopanda** [comment]
@lomamech i am still waiting for travis

**Comment by mbayopanda** [comment]
@lomamech les modifications n'ont pas encore été apporté dans la fonction `getLotsDepot` parceque le rapport des articles en stock ne calcule toujours pas correctement

**Comment by lomamech** [comment]
> @lomamech les modifications n'ont pas encore été apporté dans la fonction `getLotsDepot` parceque le rapport des articles en stock ne calcule toujours pas correctement

Yes I just added it and for this function, I included the value in the params

image

````
function getLotsDepot(depotUuid, params, finalClause) {
  let _status;
  // token of query to add if only no empty lots should be returned
  let excludeToken = '';

  if (depotUuid) {
    params.depot_uuid = depotUuid;
  }

  if (params.status) {
    _status = params.status;
    delete params.status;
  }

  if (Number(params.includeEmptyLot) === 0) {
    excludeToken = 'HAVING quantity > 0';
    delete params.includeEmptyLot;
  }

  const sql = `
    SELECT BUID(l.uuid) AS uuid, l.label, l.initial_quantity,
      SUM(m.quantity * IF(m.is_exit = 1, -1, 1)) AS quantity,
      d.text AS depot_text, l.unit_cost, l.expiration_date,
      ROUND(DATEDIFF(l.expiration_date, CURRENT_DATE()) / 30.5) AS lifetime,
      BUID(l.inventory_uuid) AS inventory_uuid, BUID(l.origin_uuid) AS origin_uuid,
      i.code, i.text, BUID(m.depot_uuid) AS depot_uuid,
      m.date AS entry_date,
      i.avg_consumption, i.purchase_interval, i.delay,
      iu.text AS unit_type,
      ig.name AS group_name, ig.expires,
      dm.text AS documentReference
    FROM stock_movement m
    JOIN lot l ON l.uuid = m.lot_uuid
    JOIN inventory i ON i.uuid = l.inventory_uuid
    JOIN inventory_unit iu ON iu.id = i.unit_id
    JOIN inventory_group ig ON ig.uuid = i.group_uuid
    JOIN depot d ON d.uuid = m.depot_uuid
    LEFT JOIN document_map dm ON dm.uuid = m.document_uuid
  `;

  const groupByClause = finalClause || ` GROUP BY l.uuid, m.depot_uuid ${excludeToken} ORDER BY i.code, l.label `;

  const filters = getLotFilters(params);
  filters.setGroup(groupByClause);

  const query = filters.applyQuery(sql);
  const queryParameters = filters.parameters();
  return db.exec(query, queryParameters)
    .then(inventories => processStockConsumptionAverage(inventories, params.dateTo, params.monthAverageConsumption))
    .then(stockManagementProcess)
    .then(processMultipleLots)
    .then((rows) => {
      if (_status) {
        return rows.filter(row => row.status === _status);
      }
      return rows;
    });
}
````

**Comment by lomamech** [comment]
@mbayopanda  watch the build function call processStockConsumptionAverage
````
return db.exec(query, queryParameters)
    .then(inventories => processStockConsumptionAverage(inventories, params.dateTo, params.monthAverageConsumption))
````

**Comment by mbayopanda** [comment]
@lomamech did you try to see the `[Stock] Article en stock` report ?

I have got this (the image below), and the calculated values of CMM are not correct 

image

**Comment by lomamech** [comment]
@mbayopanda  it's done

**Comment by lomamech** [comment]
I tested with the Tshikaji database, the CMM is calculated over 12 months
image

image

**Comment by mbayopanda** [comment]
@lomamech 👍 i tested it also the result is awesome 👍 Thank you

**Review by mbayopanda — APPROVED** [review]
LGTM 👍 

bors r+

**Pull request merged**